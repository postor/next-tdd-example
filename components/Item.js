export default ({ data }) => {
  const { html_url, full_name, owner } = data
  return (<div>
    <a href={html_url} target="_blank">{full_name}</a>
    <span>by</span>
    <a href={owner.html_url} target="_blank">{owner.login}</a>
  </div>)
}

