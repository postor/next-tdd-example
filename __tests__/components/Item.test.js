/* eslint-env jest */

import React from 'react'
import { render } from 'react-testing-library'

import Item from '../../components/Item'
import data from './item'

describe('With React Testing Library', () => {
  it('Shows "postor"', () => {
    const { getByText } = render(<Item data={data}/>)

    expect(getByText('postor')).not.toBeNull()
  })
})


describe('With React Testing Library Snapshot', () => {
  it('Should match Snapshot', () => {
    const { asFragment } = render(<Item data={data} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
