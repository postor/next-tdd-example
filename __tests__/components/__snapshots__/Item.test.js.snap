// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With React Testing Library Snapshot Should match Snapshot 1`] = `
<DocumentFragment>
  <div>
    <a
      href="https://github.com/postor/next-tdd-example"
      target="_blank"
    >
      postor/next-tdd-example
    </a>
    <span>
      by
    </span>
    <a
      href="https://github.com/postor"
      target="_blank"
    >
      postor
    </a>
  </div>
</DocumentFragment>
`;
